### ISSUE TITLE
Update S3881: Rule should allow abstract IDisposable implementations

### ISSUE THREAD

**Issue by valhristov** [SonarSource/sonar-csharp#1460]
### Description

The following code should not raise issues:

```
public abstract class DisposableBase : IDisposable
{
    public void Dispose()
    {
        Dispose(true);
        GC.SuppressFinalize(this);
    }

    protected abstract void Dispose(bool disposing);
}
```

because it is equivalent to:
```
public abstract class DisposableBase : IDisposable
{
    public void Dispose()
    {
        Dispose(true);
        GC.SuppressFinalize(this);
    }

    protected virtual void Dispose(bool disposing)
	{
	}
}
```
